File: Python/2019/Yttrium2019/SolutionTest.py
import unittest
from Solution import solution

class Test_SolutionTest(unittest.TestCase):
    def test_1(self):
        self.execute("abaacbca", 2, 3)
    def test_2(self):
        self.execute("aabcabc", 1, 5)
    def test_3(self):
        self.execute("zaaaa", 1, 1)
    def test_4(self):
        self.execute("aaaa", 2, -1)
    def test_5(self):
        self.execute("aaaa", 0, 4)
    def execute(self, s, k, expectedResult):
        actualResult = solution(s, k)
        self.assertEqual(expectedResult, actualResult)

if __name__ == '__main__':
    unittest.main()

File: 2020/Palladium2020/src/SolutionTest.java
import org.junit.Test;
import org.junit.Assert;

public class SolutionTest {

    @Test
    public void test1() {
        testInternal(10, new int[]{3, 1, 4});
    }

    @Test
    public void test2() {
        testInternal(17, new int[]{5, 3, 2, 4});
    }

    @Test
    public void test3() {
        testInternal(19, new int[]{5, 3, 5, 2, 1});
    }

    @Test
    public void test4() {
        testInternal(35, new int[]{7, 7, 3, 7, 7});
    }

    @Test
    public void test5() {
        testInternal(30, new int[]{1, 1, 7, 6, 6, 6});
    }

    private void testInternal(int expectedResult, int[] h) {
        var actualResult = new Solution().solution(h);
        Assert.assertEquals(expectedResult, actualResult);
    }
}